### ISSUE TITLE
use `export cosnt creator` instead of `export default`

### ISSUE THREAD

**Issue by azu** [secretlint/secretlint#190]
We will change rule definition.

## Changes

Secretlint rule should exports next properties.

- `messages`: messsages is messages id and value
- `creator`: a rule implementation

### Remove

Remove to support `export default creator` and `module.exports = creator`.
It will be replaced by named export.

## Motivation

This changes is strange behavior of dynamic import in CommonJS.

```
❯ cat export.cjs
module.exports = 1;

❯ cat export.mjs
export default 1;

❯ cat export_transpiled.cjs
Object.defineProperty(exports, '__esModule', { value: true })
module.exports.default = 1;
```

import above modules by dynamic import.

```js
import("./export.cjs").then(cjs => {
    console.log("cjs", cjs);
});
import("./export_transpiled.cjs").then(cjs => {
    console.log("cjs_transpiled", cjs);
});
import("./export.mjs").then(mjs => {
    console.log("mjs", mjs);
});

/*
mjs [Module: null prototype] { default: 1 }
cjs [Module: null prototype] { default: 1 }
cjs_transpiled [Module: null prototype] { __esModule: true, default: { default: 1 } }
*/
```

`cjs_transpiled` result is very strange but it is expected.

We need to use `.default.default` to import transpiled code by tsc or babel.
#187  is implemented align that.

However, we can chagne the rule interface to use named export.

If a rule use only named exports, we do not worry about the `default.default` property.

```js
Object.defineProperty(exports, '__esModule', { value: true })
module.exports.a = "a";
module.exports.default = 1;
```

```
cjs_transpiled [Module: null prototype] {
  __esModule: true,
  a: 'a',
  default: { a: 'a', default: 1 }
}
```

**Comment by azu** [comment]
Demo code: https://github.com/azu/dynamic-import-commonjs
